WcaOnRails/app/views/registrations/update_all.js.erb

```diff
@@ -0,0 +1,40 @@
+// Show flash messages
+$('#flash-alerts').replaceWith('<%=j render 'layouts/flash' %>')
+$('#flash-alerts').addClass('sticky');
+setTimeout(function() { $('#flash-alerts').fadeOut(300); }, 2500);
+
+// Update the displayed count of pending registrations
+var pendingRegistrationsCount = <%= @competition.registrations.pending.count %>;
+$('#pending-registrations-count').html(pendingRegistrationsCount);
+$('#pending-registrations-count').removeClass("label-danger label-success")
+                                 .addClass(pendingRegistrationsCount > 0 ? "label-danger" : "label-success");
+
+// Update the tables
+var $registrationsTable = $('.registrations-table:not(.floatThead-table)');
+var $pendingTable = $registrationsTable.filter('.pending');
+var $acceptedTable = $registrationsTable.filter('.accepted');
+
+var selectedRows = $pendingTable.bootstrapTable('getSelections').concat($acceptedTable.bootstrapTable('getSelections'));
+var selectedRowIds = selectedRows.map(function(row) { return row._id }); // row._id contains <tr> id
+
+$registrationsTable.each(function() {
+  $(this).bootstrapTable('remove', {
+    field: '_id',
+    values: selectedRowIds,
+  });
+});
+
+<% case params[:registrations_action] %>
+<% when "accept-selected" %>
+  $acceptedTable.bootstrapTable('append', selectedRows);
+<% when "reject-selected" %>
+  $pendingTable.bootstrapTable('append', selectedRows);
+<% end %>
```

Comment:
I get uncomfortable when I see a switch statement without a default case. Can we throw a javascript exception in the else case here?